janus_server/src/datastore.rs

```diff
@@ -615,6 +615,63 @@ impl<C: Clock> Transaction<'_, C> {
             .collect::<Result<Vec<Nonce>, Error>>()
     }
 
+    /// get_unaggregated_client_report_nonces_by_collect_for_task returns pairs of nonces and
+    /// aggregation parameters, corresponding to client reports that have not yet been aggregated,
+    /// or not aggregated with a certain aggregation parameter, and for which there are collect
+    /// jobs, for a given task.
+    #[cfg(test)]
+    #[tracing::instrument(skip(self), err)]
+    pub async fn get_unaggregated_client_report_nonces_by_collect_for_task<
```

Comment:
I don't love this method name. My attempt at doing better is `get_unaggregated_client_report_nonces_with_pending_collect_jobs_for_task`. However I think this points at a bigger nomenclature problem that we have, which is that we haven't spelled out the whole state machine for reports, aggregation jobs, collect jobs, and aggregate shares, so "pending" is kind of ambiguous there. We should probably stick with what you've got until we spell out a better state machine taxonomy and then revisit this.